src/ConnectionManager.ts

```diff
@@ -29,14 +33,17 @@ export class ConnectionManager {
     providerType: ProviderType,
     chainId: ChainId = ChainId.ETHEREUM_MAINNET
   ): Promise<ConnectionResponse> {
-    this.setConnectionData(providerType, chainId)
     this.connector = this.buildConnector(providerType, chainId)
 
+    this.connector.on(DEACTIVATE_EVENT_NAME, this.handleWeb3ReactDeactivate)
+
     const {
       provider,
       account
     }: ConnectorUpdate = await this.connector.activate()
 
+    this.setConnectionData(providerType, chainId)
```

Comment:
Moved downward to prevent storing the connection data before the connection is made. This was causing some undesired behavior were the Wallet Connect modal opened, the user closed it and when refreshing, the modal would just pop up automatically.